branding_company/model/stock_picking.py

```diff
@@ -26,19 +25,31 @@ class StockPicking(models.Model):
     """Extend stock.picking with branding."""
     _inherit = 'stock.picking'
 
-    def _prepare_invoice(
-            self, cr, uid, picking, partner, inv_type, journal_id,
-            context=None):
+    def _compute_branding(self):
+        """Branding for a delivery order is taken from sales order.
+
+        We use the procurement group to find all relates sale orders.
+        The first sale order found will determine the branding.
+        """
+        sale_model = self.env['sale.order']
+        sale_orders = sale_model.search([
+            ('procurement_group_id', '=', self.group_id)])
+        return (
+            sale_orders and
+            sale_orders[0].branding_company and
+            sale_orders[0].branding_company.id or False
```

Comment:
sale_orders[0].branding_company.id already returns False in the new API if branding_company_id is not defined.